log4j-api/src/main/java/org/apache/logging/log4j/util/EnvironmentPropertySource.java

```diff
@@ -38,21 +38,11 @@ public int getPriority() {
 		return DEFAULT_PRIORITY;
 	}
 
-	private void logException(SecurityException e) {
-		// There is no status logger yet.
-		LowLevelLogUtil.logException(
-				"The system environment variables are not available to Log4j due to security restrictions: " + e, e);
-	}
-
 	@Override
 	public void forEach(final BiConsumer<String, String> action) {
 		final Map<String, String> getenv;
-		try {
-			getenv = System.getenv();
-		} catch (final SecurityException e) {
-			logException(e);
-			return;
-		}
+		getenv = System.getenv();
+
```

Comment:
`forEach` is an optional method for a property source: Log4j2 does **not** need access to all environment variables, but if it does, it can cache the variable values.

I would leave the `catch`, so that Log4j2 users do not need to give the `getenv.*` permission to `log4j-api`.